### PR TITLE
Remove unsafe docker capabilities

### PR DESCRIPTION
**Pull request by lfitchett**
This removes 2 capabilities that docker enables by default: CAP_CHOWN and CAP_SETUID. These capabilities can be used to write to secure files on the host device and potentually gain root, as outlined here: https://labs.f-secure.com/blog/helping-root-out-of-the-container/.

These capabilities can be manually re-enabled using the normal docker CapAdd in CreateOptions.

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet.Docker/CombinedEdgeletConfigProvider.cs [hunk 1]
```diff
@@ -139,5 +143,44 @@ void MountSockets(IModule module, CreateContainerParameters createOptions)
...
             }
         }
+
+        void SetCapabilities(CreateContainerParameters createOptions)
+        {
+            // These capabilities are provided by default and can be used to gain root access:
+            // https://labs.f-secure.com/blog/helping-root-out-of-the-container/
+            HashSet<string> capabilitiesToRemove = new HashSet<string> { "CAP_CHOWN", "CAP_SETUID" };
+
+            // If customer manually adds the capabilites, don't drop them.
```
Comment (gordonwang0) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet.Docker/CombinedEdgeletConfigProvider.cs:153]: Typo: capabilities

Also, where do these capabilities come from if not the customer? Who else can set them?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet.Docker/CombinedEdgeletConfigProvider.cs [hunk 2]
```diff
@@ -139,5 +143,44 @@ void MountSockets(IModule module, CreateContainerParameters createOptions)
...
+            if (createOptions.HostConfig?.CapAdd != null)
```
Comment (gordonwang0) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet.Docker/CombinedEdgeletConfigProvider.cs:154]: Couldn't all this be simplified to something like:

```
createOptions.HostConfig.CapDrop =
    new List<string> { "CAP_CHOWN", "CAP_SETUID" }.except(createOptions.HostConfig.CapAdd);
```

(This is just from a quick glance at it, please correct me if I'm wrong)

**Review by varunpuranik — CHANGES_REQUESTED** [review]
As discussed, we need to put this behind a flag..

**Comment by lfitchett** [comment]
Replaced by #5126

**Pull request closed without merging**